### PR TITLE
correct view naming and usage

### PR DESCRIPTION
**Pull request by BenjaminW3**
This cleans up some of the parts that are mentioned in #106.

* `buf\BufPlainPtrWrapper` -> `view\ViewPlainPtr` 
* `buf\BufStdContainers` -> `view\ViewStdContainers` 
* `view\ViewBasic` -> `view\ViewSubView`
* renaming many occurences of `TBuf`, `buf`, ... to `TView`, `view`, ...
* remove the currently useless `createView` method
* Adds a `CMakeLists.txt` that builds all examples and tests at once

### CODE REVIEW

File: example/bufferCopy/CMakeLists.txt [hunk 1]
```diff
@@ -43,3 +43,6 @@ file(GLOB SRCFILES "${CMAKE_CURRENT_SOURCE_DIR}/src/*.cpp")
 
 alpaka_add_executable(${bufferCopy} ${SRCFILES})
 target_link_libraries(${bufferCopy} ${LIBS})
+
+# Group the targets into subfolders for IDEs supporting this.
+SET_TARGET_PROPERTIES(${bufferCopy} PROPERTIES FOLDER "example")
```
Comment (erikzenker) [inline example/bufferCopy/CMakeLists.txt]: The rest of the cmake methods in this file were written in small letters.

File: example/helloWorld/CMakeLists.txt [hunk 2]
```diff
@@ -43,3 +43,6 @@ file(GLOB SRCFILES "${CMAKE_CURRENT_SOURCE_DIR}/src/*.cpp")
 
 alpaka_add_executable(${helloWorld} ${SRCFILES})
 target_link_libraries(${helloWorld} ${LIBS})
+
+# Group the targets into subfolders for IDEs supporting this.
+SET_TARGET_PROPERTIES(${helloWorld} PROPERTIES FOLDER "example")
```
Comment (erikzenker) [inline example/helloWorld/CMakeLists.txt]: The rest is in small letters.

**Comment by erikzenker** [comment]
Only found disagreement in coding style in the cmake files. If we could get an agreement on that then  I would merge this.

**Comment by BenjaminW3** [comment]
fixed

**Comment by erikzenker** [comment]
Well done :high_brightness:

**Pull request merged**